examples/threathunter/threat_intelligence/stix_taxii.py

```diff
@@ -0,0 +1,270 @@
+        except handled_exceptions as e:
+            logging.warning(f"problem polling taxii server: {e}")
+        return content_blocks
+
+    def parse_collection_content(self, content_blocks):
+        for block in content_blocks:
+            yield from parse_stix(block.content, self.config.default_score)
+
+    def import_collection(self, collection):
+        num_times_empty_content_blocks = 0
+        advance = True
+        reports_limit = self.config.reports_limit
+        logging.info(f"reports limit: {reports_limit}")
+        feed_helper = FeedHelper(self.config.start_date,
+                                 self.config.minutes_to_advance)
+
+        # config parameters `start_date` and `minutes_to_advance` tell this Feed Helper
+        # where to start polling in the collection, and then will advance polling in chunks of
+        # `minutes_to_advance` minutes until we hit the most current `content_block`.
+        while feed_helper.advance():
+            num_reports = 0
+            content_blocks = self.poll_server(collection, feed_helper)
+            reports = self.parse_collection_content(content_blocks)
+            for report in reports:
+                yield report
+                num_reports += 1
+                if reports_limit is not None and num_reports > reports_limit:
+                    logging.info(f"Reports limit of {reports_limit} reached")
+                    advance = False
+                    break
+
+            if not advance:
+                break
+            if collection.type == 'DATA_SET':  # data is unordered, not a feed
+                logging.info(f"collection:{collection}; type data_set; breaking")
+                break
+            if num_reports == 0:
+                num_times_empty_content_blocks += 1
+            if num_times_empty_content_blocks > self.config.fail_limit:
+                logging.error('Max fail limit reached; Exiting.')
+                break
+            if reports_limit is not None:
+                reports_limit -= num_reports
+
+    def import_collections(self, available_collections):
+        desired_collections = self.config.collections
+        desired_collections = [x.strip()
+                               for x in desired_collections.lower().split(',')]
+
+        want_all = False
+        if '*' in desired_collections:
+            want_all = True
+
+        for collection in available_collections:
+            if collection.type != 'DATA_FEED' and collection.type != 'DATA_SET':
+                logging.debug(f"collection:{collection}; type not feed or data")
+                continue
+            if not collection.available:
+                logging.debug(f"collection:{collection} not available")
+                continue
+            if want_all or collection.name.lower() in desired_collections:
+                yield from self.import_collection(collection)
+
+    def generate_reports(self):   # NOTE:ignoring binary for now
```

Comment:
We Know it's a Note, no reason to specify in the comment.

Reply:
Sometimes when I want to flag something as being particularly important, I'll preface the comment with "N.B." (Latin _nota bene,_ "note well"). Maybe this is a case that requires that.

Reply:
Comments should be few and far between, so any comment should be viewed in a `Nota Bene` light by default. Calling it out in any way, NOTE:/TODO:/etc is just redundant.  [The google python docstring & comment standards](http://google.github.io/styleguide/pyguide.html#38-comments-and-docstrings) do a reasonably good job of explaining why those qualifiers on a comment are unnecessary.

Reply:
The comment was no longer relevant to ThreatIntel -- removed